app/src/main/java/com/therap/amin/currencyconverter/MainActivity.java

```diff
@@ -56,25 +50,19 @@ public void beforeTextChanged(CharSequence s, int start, int count, int after) {
             public void onTextChanged(CharSequence s, int start, int before, int count) {
                 if (!s.toString().equals("")) {
                     double convertedAmount = Double.parseDouble(s.toString());
-                    float conversionVal = sharedPreferences.getFloat(key,0);
+                    float conversionVal = sharedPreferences.getFloat(Constants.conversionRateKey, 0);
```

Comment:
as shared preference don't allow to put double, you can put conversionVal as string there and parse it as double after reading from shared preference.